cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -581,13 +586,57 @@ public UserTaskInfo setState(TaskState nextState) {
       return this;
     }
 
+    public CruiseControlParameters parameters() {
+      return _parameters;
+    }
+
     boolean isUserTaskDone() {
       return _futures.get(_futures.size() - 1).isDone();
     }
 
     boolean isUserTaskDoneExceptionally() {
       return _futures.get(_futures.size() - 1).isCompletedExceptionally();
     }
+
+    @Override
+    public String toString() {
+      return String.format("UserTask [%s] for endpoint [%s] with requestUrl [%s] started-at [%s]. UserTask has %d step(s).",
+                           userTaskId().toString(), endPoint().toString(), requestWithParams(),
+                           KafkaCruiseControlUtils.toDateString(startMs(), DATE_FORMAT, TIME_ZONE),
+                           futures().size());
+    }
+  }
+
+  // Persist user request and result to log. Called when the user task completes and is moved into COMPLETED status
+  private void persistCompletedUserTask(UserTaskInfo userTaskInfo) {
+    String userTaskId = userTaskInfo.userTaskId().toString();
+    List<OperationFuture> futures = userTaskInfo.futures();
+
+    RESULT_LOG.info(userTaskInfo.toString());
```

Comment:
can we also put this message into `StringBuilder` first?

Reply:
talked offline, this LOG will create new line and it is more friendly to search.